perf(core/rust/ui): use 16-bit coordinates instead of 32

mmilata opened the pull request:
Some layouts store a bunch of rectangles on the heap. This PR makes e.g. PinDialog (~12 buttons) smaller by 150 bytes.

@jpochyla can you see any reason not to do this?

jpochyla commented on core/embed/rust/src/trezorhal/display.rs:
```diff
@@ -3,8 +3,8 @@ use core::ptr;
-    pub width: u16,
-    pub height: u16,
+    pub width: i16,
+    pub height: i16,
```
Unsigned dimensions would be a net negative in the end

matejcik replied:
care to elaborate?

might be the case in general, but specifically here we're talking about TOIF image dimensions which are encoded as `u16`

jpochyla replied:
But you mentioned "all cases" 🤷 Once you add/subtract/whatever them from a signed values (and you always want to have these), it's best to leave them signed as well, even if you think dimensions should never be negative. It does not have any positive impact to have them unsigned.

mmilata replied:
Reverted ToifInfo to u16 in [cd8a6b0](https://github.com/trezor/trezor-firmware/pull/2489/commits/cd8a6b044daa581ff37adbf2e15e00b6ce69fd2f). Also moved most of the casts to trezorhal/display.rs.

matejcik commented:
worth considering: change argument type in `trezorhal::display` instead of `ui::display`

prusnak commented:
Does it make sense to change the font from i32 to i16/u16?

matejcik reviewed (approved):
LGTM, feel free to merge or implement @prusnak's suggestion.

the font id was signed for some reason -- is it because the original impl used positive numbers as colors and negative as fonts? we don't need that and we could probably get away with just `u8` for font --- or `repr(u8)` enum for that matter?

mmilata commented:
Rebased to master and changed Font to `repr(u8)` enum in [f4d958e](https://github.com/trezor/trezor-firmware/pull/2489/commits/f4d958ea1b4bddfafd16b0fe73d8313f7e7dc1b6).

matejcik commented:
@TychoVrahe are there some results from your font reorg work?

TychoVrahe commented:
@matejcik the font reorg is included in #2471

matejcik commented:
LGTM now.
